### DISCUSSION TITLE
Maven dependency needed with -javagent ? |spring boot

### DISCUSSION THREAD

**Discussion by PandeoF1** [open-telemetry/opentelemetry-java-instrumentation#10987] [Q&A] [answered]
Hi
I'd like to know if adding -javagent=opentelemetry.jar to my Spring Boot application eliminates the need to add the Maven dependencies for OpenTelemetry.

```xml
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-bom</artifactId>
            <version>1.36.0</version>
            <type>pom</type>
            <scope>import</scope>
        </dependency>
        <dependency>
            <groupId>io.opentelemetry.instrumentation</groupId>
            <artifactId>opentelemetry-instrumentation-bom-alpha</artifactId>
            <version>2.2.0-alpha</version>
            <type>pom</type>
            <scope>import</scope>
        </dependency>
         <dependency>
             <groupId>io.opentelemetry.instrumentation</groupId>
	     <artifactId>opentelemetry-spring-boot-starter</artifactId>
	</dependency>
        ```

**Comment by steverao** [comment]
Yes, if you use Java Agent approach and you don't need to add any dependency.

**Reply by PandeoF1** [reply]
So ?
Did I need to add dependencies for spring boot if i use the agent.
(Do I will have a difference on the traces between agent and deps?)

**Reply by steverao** [reply]
No, library and java agent, you can think of them as equivalent, just choose an appropriate one. For more detail you can refer to https://opentelemetry.io/docs/languages/java/automatic/spring-boot/#how-to-instrument-spring-boot-with-opentelemetry

**Reply by laurit** [reply, marked as answer]
> Did I need to add dependencies for spring boot if i use the agent.

no you don't need to add additional dependencies

> Do I will have a difference on the traces between agent and deps?

There may be differences, agent has more instrumentation than what is available in the standalone instrumentation libraries.
